regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/MultipleWhitespaceFinder.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * SonarSource Analyzers Regex Parsing Commons
+ * Copyright (C) 2009-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.analyzer.commons.regex.finders;
+
+import java.util.ArrayDeque;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.regex.Pattern;
+import org.sonarsource.analyzer.commons.regex.RegexIssueReporter;
+import org.sonarsource.analyzer.commons.regex.RegexSource;
+import org.sonarsource.analyzer.commons.regex.ast.CharacterTree;
+import org.sonarsource.analyzer.commons.regex.ast.IndexRange;
+import org.sonarsource.analyzer.commons.regex.ast.RegexBaseVisitor;
+import org.sonarsource.analyzer.commons.regex.ast.RegexSyntaxElement;
+import org.sonarsource.analyzer.commons.regex.ast.RegexTree;
+import org.sonarsource.analyzer.commons.regex.ast.SequenceTree;
+
+public class MultipleWhitespaceFinder extends RegexBaseVisitor {
+
+  private static final String MESSAGE = "Replace spaces with quantifier `{%s}`.";
+
+  private final RegexIssueReporter.ElementIssue regexElementIssueReporter;
+
+  public MultipleWhitespaceFinder(RegexIssueReporter.ElementIssue regexElementIssueReporter) {
+    this.regexElementIssueReporter = regexElementIssueReporter;
+  }
+
+  @Override
+  public void visitSequence(SequenceTree sequence) {
+
+    if (!sequence.activeFlags().contains(Pattern.COMMENTS)) {
+      Deque<RegexTree> whitespaces = new ArrayDeque<>();
+      for (RegexTree item : sequence.getItems()) {
+        if (isWhitespace(item)) {
+          whitespaces.add(item);
+        } else if (whitespaces.size() > 1) {
+          reportMultipleWhitespaces(sequence, whitespaces);
```

Comment:
I think you should have `whitespaces.clear()` right after line 54.
Don't you?

Reply:
I did the clear in the reportMultipleWhitespaces method. But I think doing it right after the call is more readable.